presto-jdbc/src/main/java/com/facebook/presto/jdbc/PrestoStatement.java

```diff
@@ -189,8 +241,15 @@ public int getUpdateCount()
             throws SQLException
     {
         checkOpen();
-        // Updates are not allowed yet so return -1
-        return -1;
+        return Ints.checkedCast(getLargeUpdateCount());
```

Comment:
Throwing `IllegalArgumentException` is definitely wrong here (this method doesn't take any arguments).

I think the two choices are `Ints.saturatedCast` which will return `Integer.MAX_VALUE` or throwing a `SQLException`. @dain and I agree that the former seems like a better choice.

Do you know what other drivers return here? For example, the Teradata driver? I remember that Netezza driver from 2010 threw an exception, which made it impossible to use JDBC for batch ETL (and this was long before `getLargeUpdateCount()` was added in Java 1.8).